src/solver.rs

```diff
@@ -199,6 +200,46 @@ pub enum Dependencies<P: Package, VS: VersionSet, M: Eq + Clone + Debug + Displa
     Available(DependencyConstraints<P, VS>),
 }
 
+/// Some statistics about how much trouble the resolver has had with a package.
+pub struct PackageResolutionStatistics {
+    discovery_order: u32,
+    conflict_count: u32,
+}
+
+impl PackageResolutionStatistics {
+    fn new<P: Package>(pid: Id<P>, conflict_count: &Map<Id<P>, u32>) -> Self {
+        Self {
+            discovery_order: pid.into_raw() as u32,
+            conflict_count: conflict_count.get(&pid).cloned().unwrap_or_default(),
+        }
+    }
+
+    /// The number of packages known by Pubgrub when this package was mentioned for the first time.
+    ///
+    /// The root package will return `0`. It's direct dependencies will start at `1` and go up from there.
+    /// Prioritizing based on this value directly will lead to a depth first search of the resolution graph.
+    /// Prioritizing based on the reverse of this value will lead to a breadth first search of the resolution graph.
+    ///
+    /// Note: The exact values depend on implementation details of PubGrub and its dependencies.
+    /// So should not be relied on and may change between any lock file update.
+    pub fn discovery_order(&self) -> u32 {
+        self.discovery_order
+    }
+
+    /// The number of times this package was involved in a conflict that caused a back jump.
+    ///
+    /// When resolution is proceeding normally, this value will stay at `0` for all packages.
+    /// Therefore, using this for prioritization will not affect the properties of simple cases
+    /// like checking a lock file.
+    /// Prioritizing based on this value directly allows the resolver to focus on the packages
+    /// it is having the most problems with.
+    ///
+    /// Note: The exact values depend on implementation details of PubGrub. So should not be relied on and may change.
+    pub fn conflict_count(&self) -> u32 {
```

Comment:
This may be very useful for something i'm writing right now :eyes:

I'm tracking two different properties atm: Which package previously chosen package A is conflicting with the currently chosen package B, because i want to deprioritize A (it's always making the latter choices fail) while prioritizing B (if i pick B first, I have a good chance at getting the right version of A, without walking through all versions of B first), which requires some mechanism of knowing which is which.